app/controller/LayerTreeController.js

```diff
@@ -202,6 +207,29 @@ Ext.define('CpsiMapview.controller.LayerTreeController', {
         });
     },
 
+
+    /**
+     * Ensures tree and map only contains layers for which
+     * the user has the required roles to see.
+     */
+    filterLayersByRole: function () {
+        var me = this;
+        var store = me.getView().getStore();
+        if (!store){
+            return;
+        }
+        store.filterBy(function(record){
+            var requiredRoles = record.get('requiredRoles');
+            if (requiredRoles && Ext.isArray(requiredRoles) && requiredRoles.length){
+                var showNode = CpsiMapview.util.RoleManager.hasAtLeastOneRequiredRole(requiredRoles);
+                // needed to ensure layer shown/hidden on both tree and map
+                record.getOlLayer().setVisible(showNode);
```

Comment:
Not sure if this should be the case - the node should become visible, but only turned on if set in default.json with the:

      "openLayers": {
        "visibility": true
      }